### PR TITLE
Dynamic Navigation

### LINKED ISSUE #29: Page navigation
We need some sort of page-navigation either in the form of a navbar or something a bit more creative, like clickable cards.
Could also be links/sidebarmenu/bottommenu. 

The more creative the better in my opinion.

### PR DESCRIPTION
**Pull request by MalteMagnussen**
~HELP: [Something here broke it. ](https://github.com/MalteMagnussen/CA2-FrontEnd/compare/613af67..d67c56a)~ | It was a hoisting issue. 

Fixes #29 

Good way to review:
- Checkout db687c4 and look at that code first. 
[](https://postimg.cc/mhK3f2Hy)
- Then checkout latest commit and look at that. 
- Maybe also watch the video: https://www.youtube.com/watch?v=xN9QxPtK2LM for the in-depth. 
It is so smart imo. 

Wish we had learned this in class.

### CODE REVIEW

File: public/style.css [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+   margin: 0;
+   font-family: Arial, Helvetica, sans-serif;
+}
+
+.container {
+   margin-top: 2em;
+   max-width: 1200px;
+   margin-left: auto;
+   margin-right: auto;
+}
```
Comment (MalteMagnussen) [inline public/style.css]: I just installed a CSS formatter extension and pressed alt+shift+f

File: public/home.html [hunk 2]
```diff
@@ -0,0 +1 @@
+<p> Welcome to Hold Krykkes homepage for Course Assignment 2. </p>
```
Comment (MalteMagnussen) [inline public/home.html]: TODO - Change these pages.

File: public/get.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<p>get.html page here. Exmaple page. add whatever you want here. Or add new pages. Just make sure 
+    to add them in the navbar also, and href it correctly. the rest should be automatic. 
+</p>
```
Comment (MalteMagnussen) [inline public/get.html]: TODO - Change these pages.

File: public/add.html [hunk 4]
```diff
@@ -0,0 +1 @@
+<p> Add "add-functionality" here I guess? We can make as many pages as we want now. These are examples. </p>
```
Comment (MalteMagnussen) [inline public/add.html]: TODO - Change these pages.

File: src/index.js [hunk 5]
```diff
@@ -16,6 +16,83 @@ document.addEventListener("DOMContentLoaded", function () {
...
+navigate();
+/* Listen for fragment identifier value changes (The # at the end of the URL is the fragment) */
+/* Navigate whenever the fragment identifier value changes */
+window.addEventListener("hashchange", navigate);
+/* Encapsulates an HTTP GET request using XMLHttpRequest
+Fetches the file at the given path, then calls the 
+callback with the content of the file. */
+function fetchFile(path, callback) {
+    // Create a new AJAX request for fetching the partial HTML file.
+    var request = new XMLHttpRequest();
```
Comment (Castau) [inline src/index.js]: Isn't this the old way of fetching? Shouldn't we be using the fetch()-function instead, as we do in the rest of the js?

Comment (MalteMagnussen) [reply src/index.js]: I put in a TODO comment about this :)

**Pull request merged**